cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -201,6 +212,35 @@ public MasterEnvironmentRunnable createRunnable(MasterEnvironmentRunnableContext
       .newInstance(context, this);
   }
 
+  @Override
+  public SparkConfig getSparkSubmitConfig() {
+    // Get k8s master path for spark submit
+    String master = getMasterPath();
+
+    // Create pod spec with cdap conf.
+    V1Pod v1Pod = new V1Pod();
+    v1Pod.setSpec(getPodSpecWithCConf(podInfo));
+
+    // Create spark template file.
+    File templateFile = new File("podTemplate");
```

Comment:
does this ever get cleaned up? Maybe SparkConfig should be Closeable so that resources like this are cleaned up?

Or do we assume submission is always in a pod so it doesn't matter?

Reply:
Yes, since it will be in the pod that will get deleted, it should not matter.